java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeSECheck.java

```diff
@@ -191,7 +195,26 @@ private static ProgramState handleDivisionRemainder(ProgramState previousState,
         // Reuse zero
         return currentState.unstackValue(1).state.stackValue(op1);
       }
-      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class).addConstraint(result, ZeroConstraint.NON_ZERO);
+      if (isNonZero(currentState, op1)) {
+        return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class).addConstraint(result, ZeroConstraint.NON_ZERO);
+      }
+      return currentState.removeConstraintsOnDomain(result, BooleanConstraint.class);
+    }
+
+    private static ProgramState handleRemainder(ProgramState previousState, ProgramState currentState) {
+      List<SymbolicValue> operands = previousState.peekValues(2);
+      SymbolicValue result = currentState.peekValue();
+      SymbolicValue op1 = operands.get(0);
+      SymbolicValue op2 = operands.get(1);
+      if (isZero(currentState, op2)) {
+        // Division by zero
+        // TODO exceptional yield ?
+        return null;
+      } else if (isZero(currentState, op1)) {
+        // Reuse zero
+        return currentState.unstackValue(1).state.stackValue(op1);
+      }
```

Comment:
up until that point this is a complete duplication of handleDivision method, maybe some factoring can be done here ?